ferry.go

```diff
@@ -292,6 +292,102 @@ func (f *Ferry) Initialize() (err error) {
 	return nil
 }
 
+func (f *Ferry) CatchUpOnMissedBinlogsOnResume() error {
+	if f.StateToResumeFrom == nil {
+		return nil
+	}
+
+	tableSchemaCache := f.StateToResumeFrom.LastKnownTableSchemaCache
+
+	// Step 1: setup the binlog reconciler
+	binlogReconciler := &BinlogReconciler{
+		BatchSize:        f.Config.BinlogEventBatchSize,
+		TableSchemaCache: tableSchemaCache,
+
+		SourceDB:     f.SourceDB,
+		BinlogWriter: f.BinlogWriter,
+	}
+	binlogReconciler.Initialize()
+
+	// TODO: Step 2: setup the iterative verifier reconciler
+
+	// Step 3: initialize the binlog streamer and the throttler
+	binlogStreamer := f.NewBinlogStreamer()
+	binlogStreamer.TableSchema = tableSchemaCache
+
+	// Step 4: attach the reconcilers' binlog event listeners
+	binlogStreamer.AddEventListener(binlogReconciler.AddRowsToStore)
+	// TODO: attach the iterative verifier reconciler binlog listener
+
+	// Step 5: Connect the binlog streamer from the resuming coordinate
+	binlogStreamer.ConnectBinlogStreamerToMysqlFrom(f.StateToResumeFrom.LastWrittenBinlogPosition)
+
+	// Step 6: Start the throttler
+	ctx, shutdown := context.WithCancel(context.Background())
+	throttlerWg := &sync.WaitGroup{}
+	throttlerWg.Add(1)
+	go func() {
+		defer throttlerWg.Done()
+
+		err := f.Throttler.Run(ctx)
+		if err != nil {
+			f.ErrorHandler.Fatal("throttler", err)
+		}
+	}()
+	defer func() {
+		shutdown()
+		throttlerWg.Wait()
+	}()
```

Comment:
why is this here? seems very out of place. smells like it should be started before we even get anywhere close to here